frontend/src/lib/bare-metal-assets.ts

```diff
@@ -133,3 +105,101 @@ export async function attachBMAs(assets: JsonArray, hosts: JsonArray, clusterNam
         }
     })
 }
+
+export async function importBMAs() {
+    return new Promise((resolve, reject) => {
+        const input = document.createElement('input')
+        input.setAttribute('id', 'importBMAs')
+        input.type = 'file'
+        input.accept = '.csv, .txt'
+        input.style.visibility = 'hidden'
+        const body = document.querySelector('body')
+        body?.appendChild(input)
+        input.onchange = (e) => {
+            input.remove()
+            const file = e.target.files[0]
+            const reader = new FileReader()
+            reader.readAsText(file, 'UTF-8')
+            reader.onload = async (readerEvent) => {
+                const content = readerEvent.target.result
+
+                // parse csv
+                setTimeout(() => {
+                    const allTextLines = content.split(/\r\n|\n/)
+                    const headers = allTextLines.shift().split(',')
+                    if (headers.length > 3) {
+                        const lines = []
+                        allTextLines.forEach((line) => {
+                            const data = line.split(',')
+                            if (data.length === headers.length) {
+                                const arr = []
+                                headers.forEach((header, inx) => {
+                                    arr.push(`"${header.trim()}": "${data[inx].trim()}"`)
+                                })
+                                lines.push(`{${arr.join(',')}}`)
+                            }
+                        })
+
+                        try {
+                            let bmas = JSON.parse(`[${lines.join(',')}]`)
+                            bmas = bmas.map((bma) => {
+                                return {
+                                    name: bma.hostName,
+                                    namespace: bma.hostNamespace,
+                                    bootMACAddress: bma.macAddress,
+                                    role: bma.role,
+                                    uid: `${bma.hostNamespace}/${bma.hostName}`,
+                                    bmc: {
+                                        address: bma.bmcAddress,
+                                        username: bma.username,
+                                        password: bma.password,
+                                    },
+                                }
+                            })
+
+                            resolve(bmas)
+                        } catch (err) {
+                            reject(err)
+                        }
+                    }
+                }, 100)
+            }
+        }
+        input.click()
+    })
+}
+
+export async function createBMAs(bmas: JsonArray, assets = [], errors = []) {
+    let results
+    let response
+    // determine the namespaces required and create them
+    const namespaces = Object.keys(keyBy(bmas, 'namespace'))
```

Comment:
We don't feel like using lodash functions is necessary because modern Javascript has most of the lodash features now.  Lodash also frequently arises in security scans.  I think temptifly is the only reason lodash is even in this repo now?

Reply:
modern js has a spec that adds lodash functions like map, etc (which babel/typscript lib actually add), however there is no modern js spec for keyBy

also the way we include lodash, we're only pulling in that function, not the whole lodash library

also it looks like James has a function that keeps lodash up to the current version